### PR TITLE
Bugfixes

### PR DESCRIPTION
**Pull request by TeToNN**
Fixes:

- Struggle no pp 
- Wormadam weird type 
- Fixing some bug in the chatroom
- Fixing avatar not showing in battle (unless id > 100)

New :
- Enabling user to join a private chatroom
- Enabling user to change avatars
- Changelog in the update viewer

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -48,9 +48,3 @@ dependencies {
     compile 'com.loopj.android:android-async-http:1.4.5'
     compile files('libs/android-integration-3.1.0.jar')
 }
-
-task appStart(type: Exec, dependsOn: 'installDebug') {
```
Comment (NamTThai) [inline app/build.gradle]: We don't need this one anymore?

File: app/src/main/java/com/pokemonshowdown/app/BattleFieldActivity.java [hunk 2]
```diff
@@ -338,19 +342,33 @@ public void run() {
...
+                                @Override
+                                public void onClick(DialogInterface dialog, int which) {
+                                    new DownloadUpdateTask(BattleFieldActivity.this).execute();
+                                }
+                            })
+                            .setNegativeButton(R.string.dialog_cancel, null)
+                            .create()
+                            .show();
+                }
+
```
Comment (NamTThai) [inline app/src/main/java/com/pokemonshowdown/app/BattleFieldActivity.java]: nice!

File: app/build.gradle (hunk 1 above)
Comment (TeToNN) [inline app/build.gradle]: Which?

On 3 August 2015 at 17:14, Nam Thai <notifications@github.com> wrote:

> In app/build.gradle
> <https://github.com/NamTThai/PokemonShowdownAndroidClient/pull/98#discussion_r36095150>
> :
>
> > @@ -48,9 +48,3 @@ dependencies {
> >      compile 'com.loopj.android:android-async-http:1.4.5'
> >      compile files('libs/android-integration-3.1.0.jar')
> >  }
> > -
> > -task appStart(type: Exec, dependsOn: 'installDebug') {
>
> We don't need this one anymore?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NamTThai/PokemonShowdownAndroidClient/pull/98/files#r36095150>
> .
>



-- 
BOUVET Clément
bouvetcl@gmail.com
 (+33) 6 32 66 03 06

Comment (TeToNN) [inline app/build.gradle]: Well i removed it and app still works fine

On 3 August 2015 at 17:14, Nam Thai <notifications@github.com> wrote:

> In app/build.gradle
> <https://github.com/NamTThai/PokemonShowdownAndroidClient/pull/98#discussion_r36095150>
> :
>
> > @@ -48,9 +48,3 @@ dependencies {
> >      compile 'com.loopj.android:android-async-http:1.4.5'
> >      compile files('libs/android-integration-3.1.0.jar')
> >  }
> > -
> > -task appStart(type: Exec, dependsOn: 'installDebug') {
>
> We don't need this one anymore?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NamTThai/PokemonShowdownAndroidClient/pull/98/files#r36095150>
> .
>



-- 
BOUVET Clément
bouvetcl@gmail.com
 (+33) 6 32 66 03 06

**Comment by TeToNN** [comment]
We can release this whenever you want (update the version first) :D

**Pull request merged**